bpf/lib/trace.h

```diff
@@ -100,6 +101,20 @@ static inline void send_trace_notify(struct __sk_buff *skb, __u8 obs_point, __u3
 		.ifindex = ifindex,
 	};
 
+	switch (obs_point) {
+		case TRACE_TO_LXC:
+		case TRACE_TO_HOST:
+		case TRACE_TO_STACK:
+		case TRACE_TO_OVERLAY:
+			update_metrics(skb->len, DIRECTION_INGRESS, REASON_FORWARDED);
+			break;
+
+		/* TRACE_FROM_LXC is handled separately in ipv*_local_delivery() */
+		case TRACE_FROM_HOST:
+		case TRACE_FROM_STACK:
+		case TRACE_FROM_OVERLAY:
+			update_metrics(skb->len, DIRECTION_EGRESS, REASON_FORWARDED);
```

Comment:
This is incorrect. TRACE_FROM_ is no indication for a successful forwarding. It can still be dropped afterwards. We may only increment REASON_FORWARD for DIRECTION_EGRESS on successful delivery.

You need to look at the call sites of `DBG_CAPTURE_DELIVERY` and increment it there.

Reply:
done